drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -128,7 +133,28 @@ function DragAndDropBlock(runtime, element) {
         });
     };
 
-    var submitLocation = function(item_id, zone, top, left) {
+    var submitLocation = function(state, item_id, zone, top, left) {
+        // If height of items wrapper has changed, adjust vertical location of item
+        var newItemsWrapperHeight = $('.items', element).height();
+
+        if (newItemsWrapperHeight < itemsWrapperHeight) {
+            var heightDelta = itemsWrapperHeight - newItemsWrapperHeight,
+                item = state.state.items[item_id],
+                newTop = parseFloat(item.top) - heightDelta,
+                itemElement = $('.option[data-value="' + item_id + '"]', element);
+
+            // Update top
+            top = newTop + 'px';
+            // Update state
+            item.top = top;
+            state.state.items[item_id] = item;
+            // Update position
+            itemElement.css('top', top);
```

Comment:
You should never manually modify DOM elements. A more "correct" way to do this would be to call `setState(state)` and have virtualDom make sure the `top` value is set properly instead, but since we already call `setState` before `submitLocation`, I feel this is getting too messy. We should always try to avoid cascading updates like this, because they make it hard to understand what's getting updated where.

It would be great if we can find a different solution, one that doesn't require to measure DOM elements. I can think of two:
- Once the item is dropped, move it from `.items` to the `.target` element, so that the coordinates are always relative to the `.target` element.
- Prevent the `.items` element from shrinking as you drag items out of it. I imagine it should be possible to create invisible (with `visibility: hidden`) duplicates of each draggable item, and position the duplicates exactly underneath the visible items. The only function of the invislble duplicates would be to maintain the dimensions of the `.items` div as you drag the visible items out. 

I suspect the first one would be easier, but I'm not entirely sure.